cmd/ci-operator-prowgen/main.go

```diff
@@ -263,9 +263,16 @@ func extractRepoElementsFromPath(configFilePath string) (*configFilePathElements
 	}
 
 	fileName := filepath.Base(configFilePath)
-	branch := strings.TrimSuffix(fileName, filepath.Ext(configFilePath))
+	s := strings.Split(strings.TrimSuffix(fileName, filepath.Ext(configFilePath)), "-")
 
-	return &configFilePathElements{org, repo, branch, fileName}, nil
+	var branchSlice []string
+	for _, v := range s {
+		if v != org || v != repo {
```

Comment:
This is still not correct. As written, the condition will be always true unless `org == repo`, extracting whole filename (sans extension). I assume you wanted to put `&&` there, but then it breaks on branch names that contain either `org` or `repo`, like this:

`openshift-prometheus-operator-openshift-master` ([this is a real example](https://github.com/openshift/prometheus-operator/tree/openshift-master)) - this would be extracted as just `master`.

I really suggest doing something like:
```
basename := strings.TrimSuffix(fileName, filepath.Ext(configFilePath))
branch := string.TrimPrefix(basename, fmt.Sprintf("%s-%s-", org, repo))
```

It is entirely possible I am overlooking some reason why this would not work, though :)

Reply:
Maybe I missed something here, but my tests say the opposite. I think I tested all the cases and this code is correct.

Reply:
My assumption is that we are renaming all ci-operator configuration files to have `$org-$repo-$branch.yaml` format. Under that assumption, a config for [openshift/prometheus-operator@openshift-master](https://github.com/openshift/prometheus-operator/tree/openshift-master) would be stored in this file:

```
ci-operator/config/openshift/prometheus-operator/prometheus-operator-openshift-master.yaml
```.

As written, this code would extract branch name incorrectly as `openshift-prometheus-operator-openshift-master` (https://play.golang.org/p/8-mQb9z-IOe)

Reply:
`ci-operator/config/openshift/prometheus-operator/master.yaml` renamed to `ci-operator/config/openshift/prometheus-operator/openshift-prometheus-operator-master.yaml`